### DISCUSSION TITLE
Call to undefined method Spatie\PdfToImage\Pdf::setCompressionQuality()

### DISCUSSION THREAD

**Discussion by temwa** [spatie/pdf-to-image#192] [Q&A]
I get the the error below when trying to compress a converted image

> Call to undefined method Spatie\PdfToImage\Pdf::setCompressionQuality()

My code is as below.

```use Spatie\PdfToImage\Pdf; // use statement is declared at the top of the file```

```
$pdf = new Pdf($file->getRealPath());
$pdf->setCompressionQuality(75);
$pdf->saveImage($storagePath . 'public/files/flipbooks/' . $savedFileName);
```

Please advise.

**Comment by MostafaAttia** [comment]
same here

**Comment by Wiejeben** [comment]
Can you check which specific package is installed? We had the same error but turned out to be `2.1.0-dev` instead of `2.1.0` probably due to a PHP requirement.

**Comment by temwa** [comment]
@Wiejeben Its strange. I seem to have 1.2 installed and when I bump it up to 2.1, I get the errors below;

>   Problem 1
    - spatie/pdf-to-image v2.x-dev requires php ^7.0 -> your php version (8.0.14) does not satisfy that requirement.
    - Root composer.json requires spatie/pdf-to-image ^2.1 -> satisfiable by spatie/pdf-to-image[2.1.0, v2.x-dev].

**Comment by Wiejeben** [comment]
Interesting, we have
```json
"spatie/pdf-to-image": "^2.1",
```

With (Of course to you it would be at least PHP 8.0)
```json
"config": {
    ....
    "platform": {
       "php": "7.4"
    }
},
```

And
```json
"minimum-stability": "dev",
"prefer-stable": true,
```

And it installed the right version.

**Comment by temwa** [comment]
I was not having the platform key inside the config value, but adding it makes no difference.

**Comment by oliverbj** [comment]
Same problem here?!
